File: Java/2D Array/Code4.java
import java.util.Scanner;

public class Code4{
    public static boolean Search(int matrix[][], int key){
        for(int i = 0; i <= matrix.length - 1; i++){
            for(int j = 0; j <= matrix[i].length - 1; j++){
                if(matrix[i][j] == key){
                    System.out.println("The data is found at ("+i+","+j+")");
                    return true;
                }
            }
        }
        System.out.println("The data is not found");
        return false;
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the number of Rows = ");
        int n = sc.nextInt();

        System.out.print("Enter the number of Columns = ");
        int m = sc.nextInt();

        int matrix[][] = new int[n][m];

        for(int i = 0; i <= matrix.length - 1; i++){
            for(int j = 0; j <= matrix[i].length - 1; j++){
                System.out.print("Enter your data for this cell = ");
                matrix[i][j] = sc.nextInt();
            }
        }
        // Print the entire matrix
        for(int i = 0; i <= matrix.length - 1; i++){
            for(int j = 0; j <= matrix[i].length - 1; j++){
                System.out.print(matrix[i][j] + " ");
            }
            System.out.println();
        }

        System.out.print("Enter the data you want to Search = ");
        int key = sc.nextInt();

        Search(matrix, key);

        sc.close();
    }
}

File: C++/2D Array/Code4.cpp
//linear search
#include<iostream>
using namespace std;

void linearSearch(int matrix[4][3], int rows, int cols, int key){//you can skip row number
    for(int i = 0; i <= rows - 1; i++){            //but you should not skip column number
        for(int j = 0; j <= cols - 1; j++){
            if(key == matrix[i][j]){
                cout << "Key found at " << "(" << i << "," << j << ")" << endl;
            }
        }
    }
}

int main(){
    int matrix[4][3] = {{1, 2, 3},{4, 5, 6},{7, 8, 9},{10, 11, 12}};
    int rows = 4;
    int cols = 3;

    int key = 8;

    linearSearch(matrix, rows, cols, key);
    return 0;
}